src/components/organisms/DataTable/DataTable.tsx

```diff
@@ -55,33 +63,49 @@ const DataTable = <TData extends object>(
     tableSize = TableSizeTypes.M,
     title,
     onSortingChange,
-    onColumnFiltersChange,
-    pagination,
-    setPagination,
+    onFiltersChange,
+    paginationData,
+    onPaginationChange,
     meta,
     getSubRows,
     pageSizeOptions,
+    hidePagination = false,
   }: DataTableProps<TData>,
   ref: Ref<HTMLDivElement>
 ) => {
+  const { per_page, total } = paginationData || {}
   const [expanded, setExpanded] = useState<ExpandedState>({})
+  const [pagination, setPagination] = useState<PaginationState>({
+    pageIndex: 0,
+    pageSize: per_page || 10,
+  })
+
+  useEffect(() => {
+    if (onPaginationChange) {
+      onPaginationChange({
+        per_page: pagination.pageSize,
+        page: pagination.pageIndex + 1,
+      })
+    }
+  }, [pagination])
```

Comment:
**[blocker]**

Hook warning here, either add the dependency or disable rule for this line. Disabling is also fine, in many cases we don't want everything in the dependencies array, but depends on the case.